templates/ordered_set.cpp

```diff
@@ -45,23 +48,64 @@ struct ordered_multiset {
 	/* unmodified functions */
 	num_t find_by_order(int k) { return (*vals.find_by_order(k)).first; }
 	int order_of_key(num_t k) { return vals.order_of_key(make_pair(k - 1, 0)); }
-	pair<num_t, int>* begin() { return vals.begin(); }
-	pair<num_t, int>* end() { return vals.end(); }
-	pair<num_t, int>* rbegin() { return vals.rbegin(); }
-	pair<num_t, int>* rend() { return vals.rend(); }
+	auto begin() { return vals.begin(); }
+	auto end() { return vals.end(); }
+	auto rbegin() { return vals.rbegin(); }
+	auto rend() { return vals.rend(); }
```

Comment:
the data type was causing some error while compiling, so just changed it to auto, and now iteration becomes easy with `for(auto x: oms)`.